plugins/woocommerce/src/Internal/ProductQueryFilters/FilterClausesGenerator.php

```diff
@@ -0,0 +1,377 @@
+			// "temp" is needed because the extra derived tables require an alias.
+			$args['where'] .= ' AND (' . join( ' temp ) AND ', $clauses ) . ' temp ))';
+		} elseif ( ! empty( $chosen_attributes ) ) {
+			$args['where'] .= ' AND 1=0';
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Get all the term for current attribute.
+	 *
+	 * @param \WP_Term[] $all_terms      Aggreated terms array.
+	 * @param string     $taxonomy       Attribute taxonomy name.
+	 * @param int        $taxonomy_count Taxonomy count. If there is only one
+	 *                                   taxonomy, return all terms.
+	 * @return mixed
+	 */
+	private function get_current_attribute_terms( $all_terms, $taxonomy, $taxonomy_count ) {
+		if ( 1 === $taxonomy_count ) {
+			return $all_terms;
+		}
+
+		return array_filter(
+			$all_terms,
+			function( $term ) use ( $taxonomy ) {
+				return $term->taxonomy === $taxonomy;
+			}
+		);
+	}
+
+	/**
+	 * Join wc_product_meta_lookup to posts if not already joined.
+	 *
+	 * @param string $sql SQL join.
+	 * @return string
+	 */
+	private function append_product_sorting_table_join( $sql ) {
+		global $wpdb;
+
+		if ( ! strstr( $sql, 'wc_product_meta_lookup' ) ) {
+			$sql .= " LEFT JOIN {$wpdb->wc_product_meta_lookup} wc_product_meta_lookup ON $wpdb->posts.ID = wc_product_meta_lookup.product_id ";
+		}
+		return $sql;
+	}
+
+	/**
+	 * If price filters need adjustment to work with displayed taxes, this returns true.
+	 *
+	 * This logic is used when prices are stored in the database differently to how they are being displayed, with regards
+	 * to taxes.
+	 *
+	 * @return boolean
+	 */
+	private function adjust_price_filters_for_displayed_taxes() {
+		$display  = get_option( 'woocommerce_tax_display_shop' );
+		$database = wc_prices_include_tax() ? 'incl' : 'excl';
+
+		return $display !== $database;
+	}
+
+	/**
+	 * Get query for price filters when dealing with displayed taxes.
+	 *
+	 * @param float  $price_filter Price filter to apply.
+	 * @param string $column Price being filtered (min or max).
+	 * @param string $operator Comparison operator for column.
+	 * @return string Constructed query.
+	 */
+	private function get_price_filter_query_for_displayed_taxes( $price_filter, $column = 'min_price', $operator = '>=' ) {
+		global $wpdb;
+
+		// Select only used tax classes to avoid unwanted calculations.
+		$product_tax_classes = $wpdb->get_col( "SELECT DISTINCT tax_class FROM {$wpdb->wc_product_meta_lookup};" );
+
+		if ( empty( $product_tax_classes ) ) {
+			return '';
+		}
+
+		$or_queries = array();
+
+		// We need to adjust the filter for each possible tax class and combine the queries into one.
+		foreach ( $product_tax_classes as $tax_class ) {
+			$adjusted_price_filter = $this->adjust_price_filter_for_tax_class( $price_filter, $tax_class );
+			$or_queries[]          = $wpdb->prepare(
+				'( wc_product_meta_lookup.tax_class = %s AND wc_product_meta_lookup.`' . esc_sql( $column ) . '` ' . esc_sql( $operator ) . ' %f )',
```

Comment:
also note that the `esc_sql` on $operator basically provides no protection. `esc_sql` only works when the escaped variable is inside quotes. It's fine here since operator is hardcoded anyways, but an allowlist would work better and signifies intent to escape.

Reply:
> should wc_product_meta_lookup be prefixed? if it is an alias, maybe we should get it from a function or a instance variable instead of using hardcoded string

We can also use `$wpdb->wc_product_meta_lookup` for explicit.  But at the same time, we [hook](https://github.com/woocommerce/woocommerce/blob/fix%2F43493-move-queryfilters-to-core/plugins/woocommerce/src/Internal/ProductQueryFilters/Controller.php#L46) into main query to inject clauses for stock status, so even we use full table name, we sill need the alias [here](https://github.com/woocommerce/woocommerce/blob/fix%2F43493-move-queryfilters-to-core/plugins/woocommerce/src/Internal/ProductQueryFilters/FilterClausesGenerator.php#L247-L247). Maybe we can create another PR to remove usage of alias in the main query?